### DISCUSSION TITLE
Bind haproxy stats endoint

### DISCUSSION THREAD

**Discussion by chuegel** [autobase-tech/autobase#405] [Q&A]
Hi,

is there a reason not to bind the haproxy stats endoint on the cluster vip if this is available?

```bash
listen stats
    mode http
{% if cluster_vip is defined and cluster_vip | length > 0 %}
    bind {{ cluster_vip }}:{{ haproxy_listen_port.stats }}
{% else %}
    bind {{ hostvars[inventory_hostname]['inventory_hostname'] }}:{{ haproxy_listen_port.stats }}
{% endif %}
    stats enable
    stats uri /
```
Edit: I meant stats not /metric

**Comment by vitabaks** [comment]
I don't understand the question

**Reply by chuegel** [reply]
I meant the stats not the /metrics :)

**Reply by vitabaks** [reply]
Anyway, I don't understand what you want us to do with the stats.

In fact, this is an optional endpoint and is not used by me.

initially, the configuration was done using the template example in the patroni repository - https://github.com/zalando/patroni/blob/master/extras/confd/templates/haproxy.tmpl#L13

**Reply by chuegel** [reply]
Right now the stats are accessible only through the cluster member IPs not the VIP (if there is any)

https://github.com/vitabaks/postgresql_cluster/blob/078d87ecac0b21c0a4ea522e01febc5ac4cf01fb/roles/haproxy/templates/haproxy.cfg.j2#L22

**Reply by vitabaks** [reply]
if cluster_vip is defined = use cluster_vip
else
use server IP adress

**Reply by chuegel** [reply]
Yes, thats what I meant. If there is a vip defined, it will be used anyway to access the master(RW)/replicas(RO) so why not for the haproxy stats also?

**Reply by vitabaks** [reply]
Please create a PR if you think this needs to be fixed
